src/python/dxpy/cli/dataset_utilities.py

```diff
@@ -0,0 +1,559 @@
+
+    dataset_id = resp['dataset']
+    out_directory = ""
+    out_file_field = ""
+    print_to_stdout = False
+    files_to_check = []
+    file_already_exist = []
+
+    if args.dump_dataset_dictionary:
+        if args.output is None:
+            out_directory = os.getcwd()
+        elif args.output == '-':
+            print_to_stdout = True
+        elif os.path.exists(args.output):
+            if os.path.isdir(args.output):
+                out_directory = args.output
+            else:
+                err_exit(fill("Error: When using -ddd, --output must be an existing directory"))
+        else:
+            err_exit(fill("Error: When using -ddd, --output must be an existing directory"))
+
+        if print_to_stdout:
+            output_file_data = sys.stdout
+            output_file_coding = sys.stdout
+            output_file_entity = sys.stdout
+        else:
+            output_file_data = os.path.join(out_directory, resp['recordName'] + ".data_dictionary" + out_extension)
+            output_file_coding = os.path.join(out_directory, resp['recordName'] + ".codings" + out_extension)
+            output_file_entity = os.path.join(out_directory, resp['recordName'] + ".entity_dictionary" + out_extension)
+            files_to_check = [output_file_data, output_file_coding, output_file_entity]
+
+    if args.fields:
+        if args.sql:
+            file_name_suffix = '.data.sql'
+        else:
+            file_name_suffix = out_extension
+        
+        if args.output is None:
+            out_directory = os.getcwd()
+            out_file_field = os.path.join(out_directory, resp['recordName'] + file_name_suffix)
+            files_to_check.append(out_file_field)
+        elif args.output == '-':
+            print_to_stdout = True
+        elif os.path.exists(args.output):
+            if os.path.isdir(args.output):
+                out_directory = args.output
+                out_file_field = os.path.join(out_directory, resp['recordName'] + file_name_suffix)
+                files_to_check.append(out_file_field)
+            else:
+                file_already_exist.append(args.output)
+        elif os.path.exists(os.path.dirname(args.output)) or not os.path.dirname(args.output):
+            out_file_field = args.output
+        else:
+            err_exit(fill("Error: {path} could not be found".format(path=os.path.dirname(args.output))))
+
+    if files_to_check:
+        for file in files_to_check:
+            if os.path.exists(file):
+                file_already_exist.append(file)
+    
+    if file_already_exist:
+        err_exit(fill("Error: path already exists {path}".format(path=file_already_exist)))
+
+    rec_descriptor = DXDataset(dataset_id, project=dataset_project).get_descriptor()
+    if args.fields is not None:
+        fields_list = ''.join(args.fields).split(',')
+        error_list = []
+        for entry in fields_list:
+            entity_field = entry.split('.')
```

Comment:
Minor: since you're only ever referring to the first and second index, you might save a bit of typing by unpacking, like:
```
a, b = entry.split(".")[:2]
```

Reply:
We are not limiting to [:2] to handle situations like `entry=entity_name.field_name.random_chars`. The code should fail and show this as an invalid input.